src/basic/copy.c

```diff
@@ -522,6 +522,12 @@ static int fd_copy_directory(
                 return -errno;
         fdf = -1;
 
+        if (copy_flags & COPY_MERGE_EMPTY) {
+                r = unlinkat(dt, to, AT_REMOVEDIR);
+                if (r < 0 && errno != ENOENT)
+                        return -errno;
+        }
```

Comment:
I don't think we should remove the target first, but reuse it, which is what i'd assume a "merge" copy would do. i.e. please add a dir_is_empty() check here... (which should probably gain a related call dir_is_empty_at() for this purpose)